### PR TITLE
[poco] iphlpapi.lib is required when building static lib on windows

### PR DESCRIPTION
**Pull request by gennesseaux**
iphlpapi.lib is required when building static lib on windows

### CODE REVIEW

File: ports/poco/fix_foundation_link.patch [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+--- a/Foundation/src/Environment_WINCE.cpp
++++ b/Foundation/src/Environment_WINCE.cpp
+@@ -23,6 +23,7 @@
+ #include <windows.h>
+ #include <iphlpapi.h>
+ 
++#pragma comment(lib, "iphlpapi.lib")
+ 
+ namespace Poco {
+
```
Comment (NancyLi1013) [inline ports/poco/fix_foundation_link.patch:24]: Could you please also check the format for this patch? Is it LF?

File: ports/poco/portfile.cmake [hunk 2]
```diff
@@ -11,7 +11,10 @@ vcpkg_from_github(
         static_pcre.patch
         # Add the support of arm64-windows
         arm64_pcre.patch
+        #
```
Comment (NancyLi1013) [inline ports/poco/portfile.cmake:14]:
```suggestion
```
Please remove this commnet.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Have you tested these features in this PR?

File: ports/poco/arm64_pcre.patch [hunk 3]
```diff
@@ -1,62 +1,62 @@
...
+--- a/Util/src/WinRegistryKey.cpp
++++ b/Util/src/WinRegistryKey.cpp
+@@ -11,6 +11,7 @@
+ // SPDX-License-Identifier:	BSL-1.0
+ //
+ 
++#pragma comment(lib,"advapi32.lib")
+ 
+ #include "Poco/Util/WinRegistryKey.h"
+ #include "Poco/Exception.h"
```
[the remark this answers is not in the document]
Comment (gennesseaux) [reply ports/poco/arm64_pcre.patch:62]: This one is `LF` not `CRLF`

File: ports/poco/fix_foundation_link.patch (hunk 1 above)
Comment (gennesseaux) [reply ports/poco/fix_foundation_link.patch]: done

File: ports/poco/portfile.cmake (hunk 2 above)
Comment (gennesseaux) [reply ports/poco/portfile.cmake]: Added comment

**Comment by gennesseaux** [comment]
Previous versions of Poco used to work on windows when building static lib.
I've updated this PR because I couldn't compile anymore and this solved my issue.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
> Have you tested these features in this PR?

I mean if you have tested these features included this PR.  Please check if `vcpkg install poco[mysql, mariadb, postgresql, sqlite3, nettle, pdf]` can work on `x86-windows` and `x64-windows-static`.

**Comment by gennesseaux** [comment]
> > Have you tested these features in this PR?
> 
> I mean if you have tested these features included this PR. Please check if `vcpkg install poco[mysql, mariadb, postgresql, sqlite3, nettle, pdf]` can work on `x86-windows` and `x64-windows-static`.

Done. everything works fine for me

**Comment by NancyLi1013** [comment]
```
Error: While reading versions for port poco from file: C:\a\1\s\versions\p-\poco.json
       Version `1.10.1#4` was not found in versions file.
       Run:

           vcpkg x-add-version poco

       to add the new port version.
```

@gennesseaux 

Sorry for the long delay.
Could you please run  `vcpkg x-add-version poco` and then submit the changes for versions to solve the failure on x86-windows?

File: ports/poco/arm64_pcre.patch (hunk 3 above)
[the remark this answers is not in the document]
Comment (NancyLi1013) [reply ports/poco/arm64_pcre.patch:62]: Sorry for the wrong comment. In order to make it more clear to review the changes, we suggest to use the original format of the patch. So would you like to update this patch?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Hi @gennesseaux 

Could you please resolve the conflicts?

**Comment by NancyLi1013** [comment]
Is work still being done for this PR @gennesseaux?

**Comment by NancyLi1013** [comment]
Thanks for the PR; we're closing this for now since there's been no response. If you'd like to continue working on it, please reopen and ping us!

**Pull request closed without merging**